atp/client.go

```diff
@@ -543,9 +565,9 @@ func (c *client) getResultV2(
 				stepData.RunID),
 		)
 	}
-	defer c.mutex.Unlock()
-	// This needs to be done after receiving the value, or else the sender will
-	// not be able to get the entry.
+	// Deletion of the entry needs to be done in this function after waiting for
+	// the value to ensure the value's lifetime is long enough in the map.
+	// It cannot be removed on the sender's side, since that would cause a race.
```

Comment:
Sounds good. Changed.
I kept the second sentence as useful context for me.

Reply:
If you really want to keep the second sentence, I would omit the part about the race because it doesn't really tell us anything.  If you really want to make a statement on this, how about something like,
> Now that we've received the result for this step, remove it from the list of running steps so that we won't see it as running anymore.  We do this here because the sender cannot tell when the message has been received, and so it cannot tell when it is safe to remove the entry from the map.

Reply:
Sure.